File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.cs
using System;
using System.Linq;

class Program
{
    static void Main()
    {
        string [] input = Console.ReadLine().Split('\n');
        string [] groups = input[0].Split('-');
        int answer = 0;
        
        
        for (int i=0; i<groups.Length; i++)
        {
            int current = groups[i].Split('+').Select(int.Parse).Sum();
            if(i==0)
            {
                answer += current;
            }
            else
            {
                answer -= current;
            }
        }
        Console.WriteLine(answer);
    }
}

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.js
let input = require('fs')
.readFileSync('/dev/stdin')
.toString().split('\n');

let groups = input[0].split('-');
let answer = 0;

for(let i=0; i<groups.length; i++){
    let current = groups[i].split('+').map(Number).reduce((a,b)=>a+b);
    if (i == 0){
        answer += current;
    }
    else{
        answer -= current;
    }
}

console.log(answer);